python_modules/dagster/dagster/_core/definitions/asset_health/asset_materialization_health.py

```diff
@@ -313,7 +332,7 @@ async def get_materialization_status_and_metadata(
     needed to power the UIs. Metadata is fetched from the AssetLatestMaterializationState object, again
     either via streamline or by computing it based on the state of the DB.
     """
-    asset_materialization_health_state = await AssetMaterializationHealthState.gen(
+    asset_materialization_health_state = await MinimalAssetMaterializationHealthState.gen(
```

Comment:
assumes that we only merge this after we know we have `MinimalAssetMaterializationHealthState` for all assets. Otherwise we could write a fallback onto `AssetMaterializationHealthState` in the case that there is no minimal state for an asset